src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressingStepType.php

```diff
@@ -42,23 +43,23 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 }
             })
             ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($options) {
-                /* @var $user UserInterface */
-                $user = $options['user'];
-                if (null === $user || !$user instanceof UserInterface) {
+                /* @var CustomerInterface $customer */
+                $customer = $options['customer'];
+                if (null === $customer || !$customer instanceof CustomerInterface) {
                     return;
                 }
 
                 /* @var $order OrderInterface */
                 $order = $event->getData();
-                if ($order->getShippingAddress() === null && $user->getShippingAddress() !== null) {
-                    $address = clone $user->getShippingAddress();
-                    $address->setUser(null);
+                if ($order->getShippingAddress() === null && $customer->getShippingAddress() !== null) {
```

Comment:
For the current codebase it is not necessary to add this check here, because in the controller there is the following line:
```
$form = $this->createCheckoutAddressingForm($order, $this->getCustomer());
```
This method creates checkout addressing form and passes customer using this method:
```
$this->getCustomer();
```
Which gets customer from currently logged user (from security context), so it will return only registered customer entity or null if user is not logged in.

Anyway I will add hasUser() method to the interface and appropriate Behat scenario to be sure that unregistered customer's address is never pre set.